onnx/checker.cc

```diff
@@ -451,34 +463,41 @@ void check_graph(
       if (output.empty()) {
         continue;
       }
-      if (output_names.count(output)) {
+
+      // TODO: should this check parent graph/s as well?
+      // Doing so could break existing models though
+      if (lex_ctx.this_graph_has(output)) {
```

Comment:
I think "lex_ctx.this_or_parent_graph_has(output)" would preserve the existing behavior, since output_names at this point, includes outer-scope names.